src/legacy/models/userHelpers.js

```diff
@@ -381,14 +382,17 @@ async function getUserGroupsToSaveAndPostMetadata(d2, api, users, existingUsersT
 }
 
 /* Save array of users (plain attributes), updating existing one, creating new ones */
-async function saveUsers(d2, users) {
+async function saveUsers(d2, users, d2Api) {
     const api = d2.Api.getApi();
+    const userRepository = new UserD2ApiRepository({ url: d2Api.baseUrl });
     const existingUsersToUpdate = await getExistingUsers(d2, {
         fields: ":owner,userCredentials,userGroups[id]",
         filter: "userCredentials.username:in:[" + _(users).map("username").join(",") + "]",
     });
     const usersToSave = getUsersToSave(users, existingUsersToUpdate);
-    return getUserGroupsToSaveAndPostMetadata(d2, api, usersToSave, existingUsersToUpdate);
+    const response = await postMetadata(api, { users: users });
+    await userRepository.getGroupsToSave(usersToSave, existingUsersToUpdate).runAsync();
+    return response;
```

Comment:
Clean arch was already poorly implemented for the whole file, and probably it's not the moment to fix it, but some notes:

- Name `getGroupsToSave` seems misleading, as it's not a get, it's an update action. Maybe `updateUserGroups`?
- I see that getUserGroupsToSaveAndPostMetadata is still used "saveCopyInUsers", is that a problem?

Reply:
Not a problem because `Copy in user` was already migrated to typescript + CA by Florian.